src/Backend/Core/Engine/Authentication.php

```diff
@@ -313,16 +316,40 @@ public static function loginUser(string $login, string $password): bool
     {
         $db = BackendModel::get('database');
 
-        // fetch the encrypted password
-        $passwordEncrypted = static::getEncryptedPassword($login, $password);
+        // build fallback for migrating devs
+        // todo remove in next major release
+        $encryptedPassword = BackendUsersModel::getEncryptedPassword($login);
```

Comment:
while this is a nice to have, won't this conflict if for instance php 5.2 or 5.3 would have a new implementation?

Also I know that the php implementation doesn't need the salt but didn't fork use a salt?

I think it would be better to just break it here and say that people will have to use the reset password (we might provide a command to trigger that)

Implementing this will make it way harder to switch internally to the symfony security component withing this mayor version and I'm not sure if the few people that will upgrade from 4 to 5 should stand in the way of this.